lib/sidekiq/processor.rb

```diff
@@ -234,5 +234,14 @@ def cloned(thing)
       Marshal.load(Marshal.dump(thing))
     end
 
+    def constantize(str)
+      names = str.split('::')
+      names.shift if names.empty? || names.first.empty?
```

Comment:
Oh, I finally got it. The first condition is to avoid calling `nil.empty?`. I'd prefer it to be more explicit though:

```ruby
names.shift if !names.empty? && names.first.empty?
```

Or

```ruby
names.shift if names.first&.empty?
```